drivers/adc/ad9250/ad9250.c

```diff
@@ -228,14 +229,15 @@ int32_t ad9250_read(struct ad9250_dev *dev,
 					 buffer,
 					 3);
 		if(ret != SUCCESS) {
-			return ret;
+			return FAILURE;
 		}
 		reg_address--;
 		reg_value <<= 8;
 		reg_value |= buffer[2];
+		*register_data = reg_value;
```

Comment:
register_data could be `NULL`. I know it sucks checking all parameters but no-OS is a library and there is no guarantee users won't pass NULL as parameter.